validator/coordinator/src/main/java/tech/pegasys/artemis/validator/coordinator/KeystoresValidatorKeyProvider.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Copyright 2020 ConsenSys AG.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+ * the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+ * an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package tech.pegasys.artemis.validator.coordinator;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+import static java.lang.String.format;
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static java.util.stream.Collectors.toList;
+import static org.apache.commons.lang3.StringUtils.isEmpty;
+import static tech.pegasys.artemis.util.alogger.ALogger.STDOUT;
+
+import com.google.common.io.Files;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Path;
+import java.util.List;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.logging.log4j.Level;
+import org.apache.tuweni.bytes.Bytes;
+import tech.pegasys.artemis.bls.keystore.KeyStore;
+import tech.pegasys.artemis.bls.keystore.KeyStoreLoader;
+import tech.pegasys.artemis.bls.keystore.KeyStoreValidationException;
+import tech.pegasys.artemis.bls.keystore.model.KeyStoreData;
+import tech.pegasys.artemis.util.bls.BLSKeyPair;
+import tech.pegasys.artemis.util.config.ArtemisConfiguration;
+import tech.pegasys.artemis.util.mikuli.KeyPair;
+import tech.pegasys.artemis.util.mikuli.SecretKey;
+
+public class KeystoresValidatorKeyProvider implements ValidatorKeyProvider {
+  static final int KEY_LENGTH = 48;
+
+  @Override
+  public List<BLSKeyPair> loadValidatorKeys(final ArtemisConfiguration config) {
+    final List<Pair<Path, Path>> keystorePasswordFilePairs =
+        config.getValidatorKeystorePasswordFilePairs();
+    checkNotNull(keystorePasswordFilePairs, "validator keystore and password pairs cannot be null");
+
+    // return distinct loaded key pairs
```

Comment:
Why does this need to be distinct? Is it an error if a user has duplicate validator keys?

Reply:
I suspect it will wind up going through a map and becoming distinct naturally anyway, but running two instances of the same validator is the quickest and easiest way to get yourself slashed, so making sure keys are distinct is a really good idea.